tests/pytest_framework/src/syslog_ng_config/statements/destinations/destination_driver.py

```diff
@@ -35,11 +35,12 @@ def __init__(self, logger_factory, native_driver_io_ref):
     def dd_read_logs(self, path, counter):
         if path not in self.__native_driver_io_collection.keys():
             native_driver_io = self.__native_driver_io_ref(self.__logger_factory, path)
-            self.__native_driver_io_collection.update({path: native_driver_io})
             native_driver_io.wait_for_creation()
             message_reader = MessageReader(
                 self.__logger_factory, native_driver_io.read, SingleLineParser(self.__logger_factory)
             )
+            self.__native_driver_io_collection.update({path: message_reader})
```

Comment:
Just a side note: I do not see why we need a hashmap of path->message_reader objects for a destination driver. Probably simply storing a single reader would be enough. At a later PR, this might be factored out.

For now, this fix is needed.